### PR TITLE
Tidy up SDK

### PR DESCRIPTION
**Pull request by silhouettes**
This was originally intended to be done later as part of productizing the SDK, however this quick PR simplifies my other work too.

Changes:

1. Clean up/de-deduplicate all of the boilerplate. The number of lines is halved and it's much more readable. I've also found out that we have some inconsistencies with how we handle response bodies in POST requests.
2. Convert everything to use template strings
3. Remove try-json dependency

### CODE REVIEW

File: sdk/package.json [hunk 1]
```diff
@@ -18,7 +18,6 @@
     "fs": "0.0.2",
     "node-uuid": "^1.4.3",
     "q": "^1.4.1",
-    "superagent": "git://github.com/visionmedia/superagent.git#45a1290b2fe5a56a1d77ca74e0b236178d58d848",
-    "try-json": "^1.0.0"
+    "superagent": "git://github.com/visionmedia/superagent.git#45a1290b2fe5a56a1d77ca74e0b236178d58d848"
```
Comment (geof90) [inline sdk/package.json]: https://github.com/visionmedia/superagent/pull/771 was merged and published to NPM, should we depend on the latest NPM version instead?

File: sdk/script/account-manager.ts [hunk 2]
```diff
@@ -9,12 +8,12 @@ var packageJson = require("../../package.json");
 
 declare var fs: any;
 
-if (typeof window === "undefined") {
-    fs = require("fs");
+if (typeof window === `undefined`) {
```
Comment (geof90) [inline sdk/script/account-manager.ts]: I'm just wondering whether we should from now on use template strings for everything, or only when it is required.

File: sdk/script/account-manager.ts [hunk 3]
```diff
@@ -55,7 +59,7 @@ export class AccountManager {
 
     public isAuthenticated(): Promise<boolean> {
         return Promise<boolean>((resolve, reject, notify) => {
-            var request: superagent.Request<any> = superagent.get(this._serverUrl + "/authenticated");
+            var request: superagent.Request<any> = superagent.get(this._serverUrl + `/authenticated`);
```
Comment (geof90) [inline sdk/script/account-manager.ts]: Why not `${this._serverUrl}/authenticated?`

File: sdk/script/account-manager.ts (hunk 2 above)
Comment (silhouettes) [inline sdk/script/account-manager.ts]: I like the idea of using it for everything, but what do you think?

The only tricky thing is with my version of the TS compiler, it doesn't seem to like it when you use template strings with ES6 style import statements.

File: sdk/package.json (hunk 1 above)
Comment (silhouettes) [inline sdk/package.json]: Yes, although it might be good to do it on a non-breaking branch, as this one won't go in for a while.

Comment (silhouettes) [inline sdk/package.json]: Ehhhh, I can just do it here I guess

File: sdk/script/account-manager.ts [hunk 4]
```diff
@@ -73,523 +77,133 @@ export class AccountManager {
...
-                            var location = res.header["location"];
-                            if (location && location.lastIndexOf("/") !== -1) {
-                                accessKey.id = location.substr(location.lastIndexOf("/") + 1);
-                                resolve(accessKey);
-                            } else {
-                                resolve(null);
-                            }
+                return this.post(`/accessKeys/`, JSON.stringify(accessKey), /*expectResponseBody=*/ false)
+                    .then((res: JsonResponse) => {
+                        var location = res.header[`location`];
```
Comment (geof90) [inline sdk/script/account-manager.ts]: : string?

Comment (geof90) [inline sdk/script/account-manager.ts]: : string?

**Comment by lostintangent** [comment]
LGTM! This is an amazing clean up job :)

File: sdk/script/account-manager.ts [hunk 5]
```diff
@@ -73,523 +77,133 @@ export class AccountManager {
...
 
-                var body = tryJSON(res.text);
-                if (res.ok) {
-                    if (body) {
-                        resolve(body.app);
-                    } else {
-                        reject(<CodePushError>{ message: "Could not parse response: " + res.text, statusCode: res.status });
-                    }
+    public addApp(appName: string): Promise<App> {
+        var app = <App>{ name: appName };
```
Comment (geof90) [inline sdk/script/account-manager.ts]: why not `var app: App = { name: appName };` ?

File: sdk/script/account-manager.ts [hunk 6]
```diff
@@ -73,523 +77,133 @@ export class AccountManager {
...
-                    var body = tryJSON(res.text);
-                    if (body) {
-                        reject(<CodePushError>body);
-                    } else {
-                        reject(<CodePushError>{ message: res.text, statusCode: res.status });
-                    }
-                }
-            });
-        });
+        var id: string = (typeof app === `string`) ? (<string><any>app) : (<App><any>app).id;
```
Comment (geof90) [inline sdk/script/account-manager.ts]: Is there a reason why this double casting is needed? Why can't we just do <string>app or <App>app?

File: sdk/script/account-manager.ts [hunk 7]
```diff
@@ -73,523 +77,133 @@ export class AccountManager {
...
-                    var body = tryJSON(res.text);
-                    if (body) {
-                        reject(<CodePushError>body);
-                    } else {
-                        reject(<CodePushError>{ message: res.text, statusCode: res.status });
-                    }
-                }
-            });
-        });
+        var id: string = (typeof deployment === `string`) ? (<string><any>deployment) : (<Deployment><any>deployment).id;
```
Comment (geof90) [inline sdk/script/account-manager.ts]: Same question as above about the double casting

**Comment by geof90** [comment]
LGTM!

File: sdk/script/account-manager.ts (hunk 6 above)
Comment (silhouettes) [inline sdk/script/account-manager.ts]: Weird. That wasn't compiling before but it does now (which makes sense). Changed it to the more logical option here

File: sdk/script/account-manager.ts (hunk 2 above)
Comment (dtivel) [inline sdk/script/account-manager.ts]: I don't think we should use template strings for everything.  As with other language features, we should use template strings when their use is appropriate and their benefits are realized.

**Comment by dtivel** [comment]
Impressive simplification.  Looks good.

**Pull request merged**